packages/@ember/array/index.ts

```diff
@@ -1892,6 +1902,167 @@ const MutableArray = Mixin.create(EmberArray, MutableEnumerable, {
 /**
 @module ember
 */
+
+type AnyArray<T> = EmberArray<T> | Array<T>;
+
+/**
+ * The final definition of NativeArray removes all native methods. This is the list of removed methods
+ * when run in Chrome 106.
+ */
+type IGNORED_MUTABLE_ARRAY_METHODS =
+  | 'length'
+  | 'slice'
+  | 'indexOf'
+  | 'lastIndexOf'
+  | 'forEach'
+  | 'map'
+  | 'filter'
+  | 'find'
+  | 'every'
+  | 'reduce'
+  | 'includes';
+
+/**
+ * These additional items must be redefined since `Omit` causes methods that return `this` to return the
+ * type at the time of the Omit.
+ */
+type RETURN_SELF_ARRAY_METHODS =
+  | '[]'
+  | 'clear'
+  | 'insertAt'
+  | 'removeAt'
+  | 'pushObjects'
+  | 'unshiftObjects'
+  | 'reverseObjects'
+  | 'setObjects'
+  | 'removeObject'
+  | 'removeObjects'
+  | 'addObject'
+  | 'addObjects'
+  | 'setEach';
+
+// This is the same as MutableArray, but removes the actual native methods that exist on Array.prototype.
+interface MutableArrayWithoutNative<T>
+  extends Omit<MutableArray<T>, IGNORED_MUTABLE_ARRAY_METHODS | RETURN_SELF_ARRAY_METHODS> {
```

Comment:
NativeArray is MutableArray with all actual native methods removed. Unfortunately, because of how Omit works, all methods that return `this` will end up not returning the correct `this`, so we also have to redefine those methods here.